akka-http-core/src/main/scala/akka/http/impl/engine/client/pool/NewHostConnectionPool.scala

```diff
@@ -114,9 +133,8 @@ private[client] object NewHostConnectionPool {
             push(responsesOut, ResponseContext(req, result))
 
         def dispatchRequest(req: RequestContext): Unit = {
-          val slot =
-            slots.find(_.isIdle)
-              .getOrElse(throw new IllegalStateException("Tried to dispatch request when no slot is idle"))
+          val slot = idleSlots.first()
```

Comment:
Are we sure there is an idle slot when we get here?

Reply:
Yes, we should. In any case, we threw before as well.